src/common.ts

```diff
@@ -409,3 +409,30 @@ export function finishNode<T extends Node>(
 
   return node;
 }
+
+
+  /** @internal */
+  export function isEqualTagName(lhs: any, rhs: any): boolean {
+    if (lhs.type !== rhs.type) {
+        return false;
+    }
+
+    if (lhs.type === 'JSXIdentifier') {
+        return lhs.name === rhs.name;
+    }
+
+    if (lhs.type === 'JSXNamespacedName') {
+      return lhs.namespace + ':' + lhs.name === rhs.namespace + ':' + rhs.name;
+    }
+
+    if (lhs.type === 'JSXMemberExpression') {
+      (
+        isEqualTagName(lhs.object, rhs.object) + '.' +
+        isEqualTagName(lhs.property, rhs.property)
+    ) === (
+      isEqualTagName(lhs.object, rhs.object) + '.' +
+        isEqualTagName(lhs.property, rhs.property)
+  )
```

Comment:
seems the (...) === (...) is useless, am I missing something?

Reply:
That code is useless at the moment, so you are right. I need to fix it and the remaining optimizing.

Reply:
@aladdin-add There are no way to extend Meriyah like you suggested. This parser is build for performance. However. I invented you to [buntis](https://github.com/buntis/buntis). 

That will be a JS & TS parser that already now accept plugins. And ofc. allow you to parse TypeScript.

Hope that solves a few issues you may have :)